### ISSUE TITLE
cosmetic defects nuki smart lock 3.0 pro?

### ISSUE THREAD

**Issue by pilot1981** [technyon/nuki_hub#141]
Hi,
I see two strange things:
- undefined NUKI LOCK state
- please configure PIN in credentials

these are related to some misconfiguration? I alreadey tried to insert PIN code, but warning don't go away....

unlock function seems to works fine...

**Comment by pilot1981** [comment]
![image](https://user-images.githubusercontent.com/46350465/223490276-fc189a7d-4648-4cf8-adbf-0fc4e6618069.png)

**Comment by technyon** [comment]
Please post the content of the "system information" page (click the firmware version number).

**Comment by pilot1981** [comment]
NUKI Hub version: 8.16
run: true
deviceId: 1329748968
mqttbroker: 192.168.6.31
mqttport: 1883
mqttuser: ***
mqttpass: ***
mqttlog: false
lockena: true
mqttpath: ingresso
openerena: false
mqttoppath: 
maxkpad: 
opmaxkpad: 
mqttca: 
mqttcrt: 
mqttkey: 
hassdiscovery: homeassistant
dhcpena: true
ipaddr: 
ipsub: 
ipgtw: 
dnssrv: 
nwhw: 2
nwhwdt: 26
rssipb: -1
hostname: ingresso
nettmout: -1
restdisc: true
resttmr: -1
rstbcn: 60
lockStInterval: 1800
configInterval: 3600
batInterval: 1800
kpInterval: 1800
kpEnabled: false
regAsApp: false
nrRetry: 3
rtryDelay: 100
crdusr: 
crdpass: 
pubauth: false
gpiolck: false
pubdbg: false
prdtimeout: 60
hasmac: false
macb0: 
macb1: 
macb2: 
MQTT connected: Yes
Lock firmware version: 
Lock hardware version: 
Lock paired: Yes
Lock PIN set: Yes
Network device: Built-in Wifi
Uptime: 174 minutes
Heap: 92948
Stack watermarks: nw: 6104, nuki: 648, pd: 236
Restart reason FW: RestartOnDisconnectWatchdog
Restart reason ESP: ESP_RST_SW: Software reset via esp_restart.

**Comment by pilot1981** [comment]
Errata corrige:

lock/unlock feature don’t work again…

This is fourth ESP32 I tried to use…I suppose this is always related to NUKI SMART LOCK 3.0 PRO model…please se other my tickets…

**Comment by mundschenk-at** [comment]
Please unpair and re-pair the ESP and the look.

**Comment by technyon** [comment]
There's nothing special about the Pro lock. Make sure to move lock and esp close together so you have a good connection. Also enable publishing rssi values and check how good the reception is.

**Comment by pilot1981** [comment]
Reset of pair solved issues…but I come back in original problem (please see other closed tickets): every day I have to reset pairing of my NUKI SMART LOCK 3.0 PRO!

**Comment by technyon** [comment]
Can you check "Register as app" before pairing again?

**Comment by pilot1981** [comment]
Please, can you explain it better?
Where I can find it?
What I have to check?

regards

**Comment by technyon** [comment]
Open the configuration portal, go to "NUKI Configuration". Under Advanced NUKI configuration, enable "Register as app" and click Save.

**Comment by pilot1981** [comment]
Register as app not solve this issue

![5CDF5147-FACC-4C96-9CFE-0D65DA330F5D](https://user-images.githubusercontent.com/46350465/224440236-aba50c4f-6262-40fe-ab2f-cc34fb209da3.png)
![3D6084EF-C611-4FED-842A-CC29E5F3F9F1](https://user-images.githubusercontent.com/46350465/224440242-b5e77e73-1212-4f53-8099-69a859609cd0.png)

**Comment by technyon** [comment]
If you use the smartphone app and check the registered users, do you see NukiHub ?

**Comment by pilot1981** [comment]
Yes

**Comment by pilot1981** [comment]
![D4650C5E-0EC9-4B33-A41C-1419E264CAA3](https://user-images.githubusercontent.com/46350465/224474778-b92d66c8-c7ab-4611-a042-a1ce13244026.png)

**Comment by pilot1981** [comment]
Anyone here is using NUKI SMART LICK 3.0 PRO VERSION?

**Comment by technyon** [comment]
This has nothing to do with  Pro or not. I know there are Pro users and it works just fine.

**Comment by pilot1981** [comment]
What I can do else to understand cause of my issue?!

**Comment by pilot1981** [comment]
What else I can do to troubleshoot this problem? 

I’m trying to solve this issue from about 1 month…

I have another NUKI SMART LOCK 3.0 working with another ESP32.

**Comment by likeyou83** [comment]
I have the same problem with 3.0 pro.
It seems that the problem occurs when connecting wifi, otherwise it works.

**Comment by pilot1981** [comment]
> yes, that worked for me.... but it's inconvenient not to be able to use the app

thanks for information: I go to try at home!

**Comment by pilot1981** [comment]
> yes, that worked for me.... but it's inconvenient not to be able to use the app.
> 
> how to connect wifi, the problem reappears

I tried to simulate same issue at home, but also with WIFI DOWN on my NUKI SMART LOCK 3.0 PRO, MQTT don’t works using NUKI hub…


I see it works only after a reboot of ESP32…but I can’t reboot it ever second…

**Comment by mundschenk-at** [comment]
> I tried to simulate same issue at home, but also with WIFI DOWN on my NUKI SMART LOCK 3.0 PRO, MQTT don’t works using NUKI hub…

What do you mean by "simulate"?

**Comment by pilot1981** [comment]
> > I tried to simulate same issue at home, but also with WIFI DOWN on my NUKI SMART LOCK 3.0 PRO, MQTT don’t works using NUKI hub…
> 
> What do you mean by "simulate"?

I shutdown WIFI connection on NUKI SMART LOCK 3.0 PRO; but issue with ESP32 is always the same

**Comment by alexdelprete** [comment]
> but issue with ESP32 is always the same

what is the issue? I read the whole thread and it's still not clear, different and possibly unrelated problems. Is it MQTT or is it the lock status/actions not working? It's two different things.

Also, did you enable publishing RSSI values? to check connection of the hub with the lock and wifi network.

**Comment by pilot1981** [comment]
The problem is NUKI SMART LOCK 3.0 PRO receive MQTT Command but give error answer…don’t lock/unlock…


Already tried to change ESP32, change physical position, retry action, enable RSSI, but this don’t solved

The only workaround found is to reboot ESP32, few seconds after ESP32 all works fine only one time, then I always receive error messages….

**Comment by alexdelprete** [comment]
The Lock doesn't receive MQTT commands, Nuki Hub processes MQTT commands and sends them to Lock via Bluetooth.

That's why it's difficult to understand where the problem is...and since there are other users with SL Pro that are working, I think the problem is how you configure things (the Hub, MQTT, or something else).

This is the 3rd time that Jan and I are asking you to enable RSSI publishing and show us the values, can you please do it so we can understand if the connection of Nuki Hub with wifi and the lock is ok?

**Comment by alexdelprete** [comment]
I also forgot to ask: are you using beta firmware on the Pro? They're testing MQTT on SL Pro, are you using that?

**Comment by pilot1981** [comment]
Ciao alessandro,
I write in English to allow all to understand  it.

I know MQTT work only between HA and ESP32…after I send MQTT command, ESP32+this firmware return to me error…

I tried it with NUKI stable firmware and beta firmware: always same result.

below you can find RSSI signal:

![801986AD-6215-4B27-95EE-BA83B8856F2C](https://user-images.githubusercontent.com/46350465/227761135-4d25bf07-e145-45b3-a9cc-957a42ef534e.jpeg)

I moved ESP32 1 meter far from NUKI SMART LICK 3.0 PRO and I’m monitoring WIFI interface using a software: not any ICMP packets lost.

I tried to do troubleshooting but this don’t solved the issue…

it’s possible to have more debug information in log file? I didn’t find any other solution…

thanks I’m advance for your help!

**Comment by technyon** [comment]
Please flash the attached debug firmware. It will generate a ton of logs, but you'll have to capture them via the serial interface, MQTT logs won't do for this. Connect the ESP to your computer and use a terminal emulator like hterm:

https://www.der-hammer.info/pages/terminal.html

Select the COM-port and connect with 115200 baud. To make the output readable you have to switch "Newline at" to "CR+LF". Perform a lock operation and capture the output ... it'll be a lot of output so maybe put it into a file and zip, so you upload it here.

Also, please post the output of the system information page (click the firmware version).

[nuki_hub-8.19-dbg-1.zip](https://github.com/technyon/nuki_hub/files/11071110/nuki_hub-8.19-dbg-1.zip)

**Comment by pilot1981** [comment]
Hi,
I installed debug firmware; this is system information page:


NUKI Hub version: 8.19-dbg-1
run: true
deviceId: 1329748968
mqttbroker: 192.168.6.31
mqttport: 1883
mqttuser: ***
mqttpass: ***
mqttlog: true
lockena: true
mqttpath: ingresso
openerena: false
mqttoppath: 
maxkpad: 
opmaxkpad: 
mqttca: 
mqttcrt: 
mqttkey: 
hassdiscovery: homeassistant
dhcpena: true
ipaddr: 
ipsub: 
ipgtw: 
dnssrv: 
nwhw: 2
nwhwdt: 26
rssipb: 60
hostname: ingresso
nettmout: -1
restdisc: true
resttmr: -1
rstbcn: 60
lockStInterval: 1800
configInterval: 3600
batInterval: 1800
kpInterval: 1800
kpEnabled: false
regAsApp: false
nrRetry: 3
rtryDelay: 100
crdusr: 
crdpass: 
pubauth: false
gpiolck: false
pubdbg: false
prdtimeout: 60
hasmac: false
macb0: 
macb1: 
macb2: 
MQTT connected: Yes
Lock firmware version: 
Lock hardware version: 
Lock paired: Yes
Lock PIN set: Yes
Lock has door sensor: No
Lock has keypad: No
Network device: Built-in Wifi
Uptime: 2 minutes
Heap: 88688
Stack watermarks: nw: 6072, nuki: 680, pd: 276
Restart reason FW: RestartOnDisconnectWatchdog
Restart reason ESP: ESP_RST_SW: Software reset via esp_restart.


I sent unlock command and this is output captured using serial connection:

Lock action received: unlock
[262689][E][NukiBle.hpp:8] executeAction(): Lock Heartbeat timeout, command failed
Lock action result: error
Lock: Last command failed, retrying after 100 milliseconds. Retry 1 of 3
[262820][E][NukiBle.hpp:8] executeAction(): Lock Heartbeat timeout, command failed
Lock action result: error
Lock: Last command failed, retrying after 100 milliseconds. Retry 2 of 3
[262951][E][NukiBle.hpp:8] executeAction(): Lock Heartbeat timeout, command failed
Lock action result: error
Lock: Last command failed, retrying after 100 milliseconds. Retry 3 of 3
[263082][E][NukiBle.hpp:8] executeAction(): Lock Heartbeat timeout, command failed
Lock action result: error
Lock: Maximum number of retries exceeded, aborting.

The only useful information seems to be the Lock Heartbeat timeout: what it means? I suppose this is internal function called using NUKI API...

**Comment by pilot1981** [comment]
I upload here complete debub logs:
[complete NUKI log.txt](https://github.com/technyon/nuki_hub/files/11072459/complete.NUKI.log.txt)

**Comment by technyon** [comment]
I think this will be useful already, but can you configure the PIN and get more logs?

**Comment by pilot1981** [comment]
I already configured PIN: I changed it in log file…

I read pin paired, it’s right?

I have to do any other things?

**Comment by pilot1981** [comment]
@alexdelprete @technyon any idea? I can do some other test?

**Comment by pilot1981** [comment]
Yes, sure!

mall works fine using NUKI APP.

I tried a soft reset…

**Comment by pilot1981** [comment]
> Check release 8.20 please. The fix for the non-random Nonce could fix your issue.

OK, installed...I go to test it...

**Comment by pilot1981** [comment]
> eMachine(): ************************ SENDING COMMAND [1] ************************
> [ 18292][D][NukiBle.cpp:829] sendEncryptedMessage(): payloadlen: 2
> [ 18298][D][NukiBle.cpp:830] sendEncryptedMessage(): sizeof(plainData): 8
> [ 18305][D][NukiBle.cpp:831] sendEncryptedMessage(): CRC: 459b
> Plain data with CRC: : 02 00 5b 38 01 00 11 00 9b 45
> Nonce: 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39
> Additional data: : 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 39 02 00 5b 38 1a 00
> Encryption key (secretKey): : 4b ca 90 d0 53 e2 21 ea 1a 1e f9 67 c7 c9 b0 2e c3 89 2a f0 4e e8 12 83 1f f4 78 7a 85 a1 b9 ab
> Plain data encrypted: : 7a f5 b8 0a 3a 0d 8f eb ea 6b 5c 63 58 0b a0 88 6b 36 48 d8 a0 0c 3b 82 cc 8c
> [ 18366][D][NukiBle.cpp:132] connectBle(): connecting within: nuki
> [ 19370][W][NukiBle.cpp:147] connectBle(): BLE Connect failed, retrying
> [ 20383][W][NukiBle.cpp:147] connectBle(): BLE Connect failed, retrying
> [ 21396][W][NukiBle.cpp:147] connectBle(): BLE Connect failed, retrying
> [ 22409][W][NukiBle.cpp:147] connectBle(): BLE Connect failed, retrying
> lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
> [ 23365][W][NukiBle.cpp:147] connectBle(): BLE Connect failed, retrying
> [ 23375][W][NukiBle.cpp:160] connectBle(): BLE Connect failed
> [ 23375][W][NukiBle.cpp:863] sendEncryptedMessage(): Send encr msg failed due to unable to connect
> [ 23378][D][NukiBle.hpp:96] cmdStateMachine(): ************************ SENDING COMMAND FAILED ************************
> Querying lock battery state: failed

Hi,
I'm using this model of ESP32:

https://it.aliexpress.com/item/1005004797369870.html?spm=a2g0o.order_list.order_list_main.29.2a7436962fyYrt&gatewayAdapt=glo2ita

I already tried to change with other 4 ESP32 devices...always same issue...I tried also to move ESP32 more close (less than 1 meter) from NUKI SMART LOCK 3.0 PRO

I used the same model to connect to another NUKI SMART LOCK 3.0: no any issue in this case...

it could be a problem of bluetooth radio interference?!

you can suggest another ESP32 model to use?

**Comment by alexdelprete** [comment]
> it could be a problem of bluetooth radio interference?!

could be. but I would first test a device that is known to be reliable.

> you can suggest another ESP32 model to use?

I use these two:

M5 Atom Lite: https://shop.m5stack.com/products/atom-lite-esp32-development-kit
M5 AtomU: https://shop.m5stack.com/products/atomu-esp32-development-kit-with-usb-a

**Comment by pilot1981** [comment]
> It's not a sw issue. You have a problem with the signal, but only you can determine if it's interference or just bad hw.

Hi,
I tried to shutdown nearest Bluetooth devices (raspberry pi3 and Apple TV 4), but this don’t  helped…

so I go to buy M5 Atom: both Atom Lite and AtomU have Bluetooth antenna? I can buy which I prefer? There is a model with best Bluetooth antenna performance?

**Comment by technyon** [comment]
If reception is the problem, maybe it's worth trying an ESP with an external antenna. The PCB antennas used on most ESPs aren't exactly great.

**Comment by mundschenk-at** [comment]
There is no difference between Atom Lite and AtomU regarding the antenna. It's not an external one unfortunately, but it's a "3D antenna" and not just some traces on the PCB. In my experience the Atom's signal strength is a bit better than with PCB antennas (but obviously not as good as an external one).

**Comment by alexdelprete** [comment]
> I can buy which I prefer? There is a model with best Bluetooth antenna performance?

They're the same, except that the AtomU is a usb stick, the other requires a usb cable for power, but it's smaller. Check the links I posted.

**Comment by pilot1981** [comment]
Ok, thanks.

I bought both devices M5…

when they arrive…I go to try …

**Comment by pilot1981** [comment]
M5 Atom arrived!

@alexdelprete please can you suggest fastest way to upload technuon firmware?

regards

**Comment by bcutter** [comment]
After this adventure https://github.com/technyon/nuki_hub/issues/176 today I'm also seeing this (randomly for two ESPs which have been erased and flashed with Nuki hub 8.22 and paired with different smart locks). PIN has been provided already plenty of times.

![grafik](https://user-images.githubusercontent.com/13799156/232249266-12afe13e-c586-491e-bec7-74fa8afde337.png)

![grafik](https://user-images.githubusercontent.com/13799156/232249277-3619fd5c-6358-41c3-9602-bde777ed81ea.png)

![grafik](https://user-images.githubusercontent.com/13799156/232249290-0412ddc3-b76d-45d2-98c6-a2f2aa9556f3.png)

![grafik](https://user-images.githubusercontent.com/13799156/232249304-e8519ac5-d41e-4dbd-8339-f4396584b160.png)

Are certain types of PINs problematic? Leading zeros or similar maybe? Just an idea.

Clearly a bug introduced somewhere between 8.8 and 8.22 😄

**Comment by pilot1981** [comment]
> After this adventure #176 today I'm also seeing this (randomly for two ESPs which have been erased and flashed with Nuki hub 8.22 and paired with different smart locks). PIN has been provided already plenty of times.
> 
> ![grafik](https://user-images.githubusercontent.com/13799156/232249266-12afe13e-c586-491e-bec7-74fa8afde337.png)
> 
> ![grafik](https://user-images.githubusercontent.com/13799156/232249277-3619fd5c-6358-41c3-9602-bde777ed81ea.png)
> 
> ![grafik](https://user-images.githubusercontent.com/13799156/232249290-0412ddc3-b76d-45d2-98c6-a2f2aa9556f3.png)
> 
> ![grafik](https://user-images.githubusercontent.com/13799156/232249304-e8519ac5-d41e-4dbd-8339-f4396584b160.png)
> 
> Are certain types of PINs problematic? Leading zeros or similar maybe? Just an idea.
> 
> Clearly a bug introduced somewhere between 8.8 and 8.22 😄

My problems seems related only to Bluetooth performance of ESP32 device

**Comment by bcutter** [comment]
My orange text is gone now. For now. Don't completely trust it yet, will have a look at it for few days.

**Comment by pilot1981** [comment]
I'm testing M5 Atom and for now all works fine!

**Comment by alexdelprete** [comment]
> I'm testing M5 Atom and for now all works fine!

Not surprised. :)

**Comment by pilot1981** [comment]
> > I'm testing M5 Atom and for now all works fine!
> 
> Not surprised. :)

Hi,
All worked fine for two days, but today problem come back!


I can flash again debug firmware 8.21 to capture detailed logs?

**Comment by alexdelprete** [comment]
> I can flash again debug firmware 8.21 to capture detailed logs?

You don't need permission to produce a log. Do it. :)

**Comment by pilot1981** [comment]
ok, this is last info using M5 atom and debug firmware:

NUKI Hub version: 8.19-dbg-1
run: true
deviceId: 2551311465
mqttbroker: 192.168.6.31
mqttport: 1883
mqttuser: ***
mqttpass: ***
mqttlog: true
lockena: true
mqttpath: ingresso
openerena: false
mqttoppath: 
maxkpad: 
opmaxkpad: 
mqttca: 
mqttcrt: 
mqttkey: 
hassdiscovery: homeassistant
dhcpena: true
ipaddr: 
ipsub: 
ipgtw: 
dnssrv: 
nwhw: 1
nwhwdt: 26
rssipb: 60
hostname: ingresso
nettmout: -1
restdisc: true
resttmr: -1
rstbcn: 60
lockStInterval: 1800
configInterval: 3600
batInterval: 1800
kpInterval: 1800
kpEnabled: false
regAsApp: false
nrRetry: 3
rtryDelay: 100
crdusr: 
crdpass: 
pubauth: false
gpiolck: false
pubdbg: false
prdtimeout: 60
hasmac: false
macb0: 
macb1: 
macb2: 
MQTT connected: Yes
Lock firmware version: 
Lock hardware version: 
Lock paired: Yes
Lock PIN set: Yes
Lock has door sensor: No
Lock has keypad: No
Network device: Built-in Wifi
Uptime: 0 minutes
Heap: 90844
Stack watermarks: nw: 6072, nuki: 456, pd: 272
Restart reason FW: NotApplicable
Restart reason ESP: ESP_RST_POWERON: Reset due to power-on event.


it worked fine for 2 days (only when NUKI smart lock PRO 3.0 WIFI was disconnect).
[log.txt](https://github.com/technyon/nuki_hub/files/11255489/log.txt)

**Comment by pilot1981** [comment]
> I have the same problem with 3.0 pro.
> It seems that the problem occurs when connecting wifi, otherwise it works.

Hi,
I notice this firmware worked fine for me when NUKI SMART LOCK 3.0 PRO have WIFI disconnected.

I connected and disconnected again WIFI, but I’m unable to replicate the problem…you can help me? I suppose this isn’t related only to WIFI disconnection…

**Comment by pilot1981** [comment]
sorry for delay: I come back with some news.

Nuki support authorized an RMA: so now I installed a new Nuki smart lock 3.0 PRO! but nuki hub using ESP32 have same problem with bluteooth!

I can confirm what @likeyou83 said: NUKI SMART LOCK 3.0 PRO is unable to mantain both active connection with bluetooth nuki bridge and built WIFI connection. I go to investigate it with NUKI support.

Anyone here with NUKI SMART LOCK 3.0 PRO can share its experience?

@alexdelprete @technyon I see a strange behaviour; when Nuki smart lock 3.0 PRO is disconnected from WIFI network, I paired ESP32 nuki hub successful and it was able to lock/unlock my door, but I don't see it in user lists of my NUKI smart lock: how It's possibile? I only see my app user...how it's possibile? I don't think app registration mode can solve my problem...


regards

**Comment by technyon** [comment]
Hi,

"Register as app" isn't really recommended to use. I just wanted to if it fixes the problem of the lock loosing the ESP. If the ESP is registered as bridge (as in not "Registered as app"), it shouldn't appear in the smartphone app.

**Comment by pilot1981** [comment]
> Hi,
> 
> "Register as app" isn't really recommended to use. I just wanted to if it fixes the problem of the lock loosing the ESP. If the ESP is registered as bridge (as in not "Registered as app"), it shouldn't appear in the smartphone app.

ok, thanks for explanation!

you can confirm NUKI SMART LOCK 3.0 PRO is unable to etablish correct communications with both nuki hub (via bluetooth with ESP32) and wifi network (via built-in WIFI interface) at the same moment? this could explain behaviour of my installation...

regards

**Comment by pilot1981** [comment]
there in answer from NUKI:

https://developer.nuki.io/t/problem-to-use-at-same-time-bluetooth-and-wifi-interface-of-nuki-smart-lock-3-0-pro/20981/3

so I can use Technyon with NUKI SMART LOCK 3.0 PRO only if I go to switch OFF built-in WIFI adpter!

@technyon you can find a workaround for this?

**Comment by alexdelprete** [comment]
> so I can use Technyon with NUKI SMART LOCK 3.0 PRO only if I go to switch OFF built-in WIFI adpter!

No. Jurgen says the wifi turns off automatically when you pair the bridge. But it seems that's not happening with the SL3 Pro.

![image](https://user-images.githubusercontent.com/7027842/235188809-4bb2ac5f-78d7-4226-91f0-c48145ae555f.png)

**Comment by mundschenk-at** [comment]
Also there are differences between the BLE API responses. This is an intentional part of their API. If you want to use Nuki Hub as a bridge, you need to register it as such.

**Comment by mundschenk-at** [comment]
No, I am suggesting that that setting causes problems and you should NOT use it.

**Comment by alexdelprete** [comment]
> you are suggesting to use feature "registered as app " in nuki huib to solve it? already tried, not solve my problem!

Why do you have the wifi turned on? I still can't understand what you are trying to achieve. Please explain what you want to do. 

I'm also suspecting that when you paired the ESP, wifi was turned automatically off (like Jurgen explained) but after 2 days you turned it on manually, and you started having problems again.

**Comment by alexdelprete** [comment]
![image](https://user-images.githubusercontent.com/7027842/235229234-4e3af679-5368-4b02-a8a4-9a8b8a1205b5.png)

You can't have both: either you use wifi or you use a bridge (that uses wifi on its own).

So the problems are most probably created by you, manually enabling wifi AFTER the pairing.

It makes NO SENSE having both. If you have a bridge, it is responsible to manage the lock through BLE. If you don't have the bridge, the SL3 Pro will use its internal wifi to be managed through the app.

These are your only options, period. Is it clear now? :)

**Comment by alexdelprete** [comment]
Official answer from Nuki devs:

![image](https://user-images.githubusercontent.com/7027842/235306122-8bc22923-3139-43a8-9b3b-f74cdc15d7ee.png)

So, keep the wifi turned off, or it won't work.

Issue can be closed now.

**Comment by pilot1981** [comment]
> > you are suggesting to use feature "registered as app " in nuki huib to solve it? already tried, not solve my problem!
> 
> Why do you have the wifi turned on? I still can't understand what you are trying to achieve. Please explain what you want to do.
> 
> I'm also suspecting that when you paired the ESP, wifi was turned automatically off (like Jurgen explained) but after 2 days you turned it on manually, and you started having problems again.

For reliability purpose I would obtain to have 2 way to control NUKI SMART LOCK 3.0 PRO:
- via Bluetooth using NUKI hub
- via WIFI using built in WIFI


So, if something fails I can use other way to control NUKI SMART 3.0 PRO!

**Comment by pilot1981** [comment]
> Official answer from Nuki devs:
> 
> ![image](https://user-images.githubusercontent.com/7027842/235306122-8bc22923-3139-43a8-9b3b-f74cdc15d7ee.png)
> 
> So, keep the wifi turned off, or it won't work.
> 
> Issue can be closed now.

Ok I suggest to add this information in wiki, so anyone can know it…

**Comment by technyon** [comment]
I'm glad NUKI finally clarified that this is by design. I've added the information to the readme, and I think the issue can be closed.
